### PR TITLE
Refs Task #402522 test model user, webhook

### PR DESCRIPTION
**Pull request by nguyenthinhhn**
## Related Tickets

- [#402522](https://dev.sun-asterisk.com/issues/402522)

## What's this PR do ?

- [x] Test model user
- [x] Test model webhook

## Performance

- [x] Resolved n + 1 query
- [x] Run explain query already
- [x] Time run rake task : 1000 ms

## Checklist

- [x] It was tested in local success?
- [x] Fill link PR into ticket and the opposite
- [x] Note purpose/reason, solution, scope of influence into ticket

## ENV note

```
```

## Notes
*(Other notes)*

### CODE REVIEW

**Comment by framgiaciv3** [comment]
coverage</br> **1. Result build** </br><table><tr><th>Style</th><th>Status</th><th>Ignored</th></tr><tr><td>yarn_install._start_run_.7</td><td>success</td><td></td></tr><tr><td>cp__env_example__env._start_run_.3</td><td>success</td><td></td></tr><tr><td>composer_install._start_run_.5</td><td>success</td><td></td></tr><tr><td>phpcs</td><td>success</td><td></td></tr><tr><td>php_artisan_config_c._start_run_.11</td><td>success</td><td></td></tr><tr><td>yarn_dev._start_run_.9</td><td>success</td><td></td></tr><tr><td>php_artisan_migrate_._start_run_.19</td><td>success</td><td></td></tr><tr><td>composer_dump-autolo._start_run_.17</td><td>success</td><td></td></tr><tr><td>phpunit</td><td>success</td><td></td></tr><tr><td>php_artisan_key_gene._start_run_.13</td><td>success</td><td></td></tr><tr><td>framgia-ci_test-conn._start_run_.15</td><td>success</td><td></td></tr></table>

**Comment by nguyenthinhhn** [comment]
ready

**Review by quanghung97 — APPROVED** [review]
OK

File: tests/Unit/Models/UserTest.php [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+
+    /**
+     * test scope search
+     *
+     * @return void
+     */
+    public function testScopeSearch()
+    {
+        $perPage = config('paginate.perPage');
+        $user = factory(User::class)->create();
```
Comment (ghost) [inline tests/Unit/Models/UserTest.php:57]: tạo thêm một só user ko thỏa mản nữa cho chắc ăn e

File: tests/Unit/Models/WebhookTest.php [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+
+    public function testScopeDisabled()
+    {
+        $webhook = factory(Webhook::class)->create(['status' => WebhookStatus::DISABLED]);
+        $result = Webhook::disable();
+
+        $this->assertEquals(1, $result->count());
+    }
+
+    public function scopeByUser($query, $userId)
```
Comment (ghost) [inline tests/Unit/Models/WebhookTest.php]: đoạn này là gì đây e

File: tests/Unit/Models/WebhookTest.php [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+            'description',
+            'room_id',
+            'room_name',
+        ];
+        $model = new Webhook();
+
+        $this->assertEquals($fillable, $model->getFillable());
+    }
+
+    public function test_webhook_be_longs_to_user()
```
Comment (ghost) [inline tests/Unit/Models/WebhookTest.php]: update tên method thành camel case cho đồng bộ e, check cho các method phía dưới luôn

File: tests/Unit/Models/WebhookTest.php [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+
+    public function scopeByUser($query, $userId)
+    {
+        return $query->where('user_id', $userId)->get();
+    }
+
+    public function testScopeSearch()
+    {
+        $perPage = config('paginate.perPage');
+        $webhook = factory(Webhook::class)->create();
```
Comment (ghost) [inline tests/Unit/Models/WebhookTest.php:114]: đoạn này cũng tạo thêm một vài webhook ko thỏa mản nữa nhé

Comment (nguyenthinhhn) [reply tests/Unit/Models/WebhookTest.php:114]: ok a

**Comment by framgiaciv3** [comment]
coverage</br> **1. Result build** </br><table><tr><th>Style</th><th>Status</th><th>Ignored</th></tr><tr><td>phpunit</td><td>success</td><td></td></tr><tr><td>yarn_dev._start_run_.9</td><td>success</td><td></td></tr><tr><td>composer_install._start_run_.5</td><td>success</td><td></td></tr><tr><td>composer_dump-autolo._start_run_.17</td><td>success</td><td></td></tr><tr><td>framgia-ci_test-conn._start_run_.15</td><td>success</td><td></td></tr><tr><td>php_artisan_migrate_._start_run_.19</td><td>success</td><td></td></tr><tr><td>php_artisan_config_c._start_run_.11</td><td>success</td><td></td></tr><tr><td>phpcs</td><td>success</td><td></td></tr><tr><td>php_artisan_key_gene._start_run_.13</td><td>success</td><td></td></tr><tr><td>cp__env_example__env._start_run_.3</td><td>success</td><td></td></tr><tr><td>yarn_install._start_run_.7</td><td>success</td><td></td></tr></table>

**Comment by framgiaciv3** [comment]
coverage</br> **1. Result build** </br><table><tr><th>Style</th><th>Status</th><th>Ignored</th></tr><tr><td>composer_dump-autolo._start_run_.17</td><td>success</td><td></td></tr><tr><td>phpcs</td><td>success</td><td></td></tr><tr><td>php_artisan_config_c._start_run_.11</td><td>success</td><td></td></tr><tr><td>framgia-ci_test-conn._start_run_.15</td><td>success</td><td></td></tr><tr><td>php_artisan_migrate_._start_run_.19</td><td>success</td><td></td></tr><tr><td>cp__env_example__env._start_run_.3</td><td>success</td><td></td></tr><tr><td>yarn_dev._start_run_.9</td><td>success</td><td></td></tr><tr><td>php_artisan_key_gene._start_run_.13</td><td>success</td><td></td></tr><tr><td>composer_install._start_run_.5</td><td>success</td><td></td></tr><tr><td>yarn_install._start_run_.7</td><td>success</td><td></td></tr><tr><td>phpunit</td><td>success</td><td></td></tr></table>

**Comment by nguyenthinhhn** [comment]
ready

**Review by ghost — APPROVED** [review]

**Review by ghost — APPROVED** [review]

**Comment by phucqd-1305** [comment]
OK

**Pull request merged**